Tests - Drop OLD committed bx-derby; install bx-derby upon build

michaelborn opened the pull request:
# Description

The BoxLang Core repo relies on the `bx-derby` module to provide a packaged version of Apache Derby for most JDBC tests. This bx-derby version has previously been committed to the repository, but this causes numerous issues:

1. The committed bx-derby version is old and out of date
2. We are not testing updated versions of bx-derby

Hence, this PR updates the gradle build to install `bx-derby` during test compilation. The bx-derby module is then loaded into the boxlang runtime via a custom function in `BaseJDBCTest.java`.

Note we have special treatment here for bx-derby as it is the only DB driver required to run JDBC tests. The MySQL tests, MSSQL tests, etc. all use `@EnableIf()` to auto-disable if the module is present; whereas the vast majority JDBC tests rely on apache derby and do not use `@EnableIf()`.

## Checklist

- [X] My code follows the style guidelines of this project [cfformat](../.cfformat.json)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] New and existing unit tests pass locally with my changes

michaelborn committed:
Tests - Drop OLD committed bx-derby; install bx-derby upon build

michaelborn committed:
PR cleanup per copilot

michaelborn committed:
Add @EnabledIf annotations for a derby-independant test suite

bdw429s commented:
The dump failure can be resolved by rebasing with dev.  There are still other failures, however.

michaelborn committed:
Drop custom derby logic from build.gradle

michaelborn committed:
Disable derby-dependant tests in non-derby contexts

lmajano commented:
@michaelborn status on this?

michaelborn committed:
Add missing EnabledIf import

michaelborn committed:
Drop custom bx-derby load from BaseJDBCTest

michaelborn committed:
Fix final derby dependencies in test suite

michaelborn committed:
Final fixes for derby-driven test suite

michaelborn committed:
Add bx-derby back to CI test setup

michaelborn commented:
Ready for review and merge, as long as this CI run passes with bx-derby installed.

michaelborn committed:
Make test scheduler less chatty

michaelborn committed:
Don't recompute datasource config unique name

If the config properties mutate, this datasoure will fail to be found in the datasource manager/service. It's quite possible we need to prevent the datasource properties from mutating as well, but that's another conversation to have.

michaelborn committed:
Add --nosave to test DB module install for contribution docs

michaelborn committed:
JDBC - Recompute datasource unique name after onTheFly and applicationName set

michaelborn committed:
Merge branch 'development' into patch/jdbc-tests-derby-switch-to-bx-derby

michaelborn committed:
JDBC - Tests - Disable hanging connection manager test

michaelborn committed:
JDBC tests - Re-enable connection manager test for windows

bdw429s committed:
refactor threaded test to not use DDL

bdw429s committed:
Missing some datasource names in MSSQL test, which is prolly using Derby on accident

michaelborn committed:
Tests - Rewrite DBInfoTest for mysql

1. Try to avoid case-sensitive database, table or column name matches as these are flakey and change from vendor to vendor.
2. Properly set up and tear down the database tables on test run init and teardown.

michaelborn committed:
Disable SQLErrorHandlingTest on windows

This continues to fail in windows CI environments.

michaelborn committed:
Add missing import for SQLErrorHandlingTest disabling on windows

michaelborn committed:
Merge branch 'development' into patch/jdbc-tests-derby-switch-to-bx-derby

michaelborn committed:
Merge branch 'development' into patch/jdbc-tests-derby-switch-to-bx-derby

bdw429s commented on src/test/bx/Scheduler.bx:
```diff
@@ -18,13 +18,6 @@ class {
 		// Setup Scheduler Properties
 		scheduler.setSchedulerName( "My-Scheduler" )
 		scheduler.setTimezone( "UTC" )
-
```
@lmajano @michaelborn Why the removals in this file?  Is this related to Derby?

michaelborn replied:
Not related to Derby, just related to the fact that the scheduler completely filled the CI log to where you couldn't find any useful test output from the junit tests.

I did get approval from Luis before doing this, see here: https://ortussolutions.slack.com/archives/C04UF3E09MH/p1762800683473899

bdw429s commented on CONTRIBUTING.md:
```diff
@@ -75,9 +77,9 @@ To run JDBC tests against real databases, you'll need to set up the dockerized d
-   1. MSSQL: `box install id=bx-mssql directory=~/.boxlang/modules`
-   2. MySQL: `box install id=bx-mysql directory=~/.boxlang/modules`
-   3. Oracle: `box install id=bx-oracle directory=~/.boxlang/modules`
+   1. MSSQL: `box install id=bx-mssql directory=~/.boxlang/modules --nosave`
```
Is there a reason not to save these?

michaelborn replied:
What would they save them to? We don't have a `box.json` file in the boxlang repo root. 

There's not a strong reason to include the `--nosave` command, but also not a great reason to create a box.json file that we haven't yet needed, IMO.
